streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateUpdateTask.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Updates the state for all Global State Stores.
+ */
+public class GlobalStateUpdateTask implements GlobalStateMaintainer {
+
+    private class SourceNodeAndDeserializer {
+        private final SourceNode sourceNode;
+        private final RecordDeserializer deserializer;
+
+        SourceNodeAndDeserializer(final SourceNode sourceNode,
+                                  final RecordDeserializer deserializer) {
+            this.sourceNode = sourceNode;
+            this.deserializer = deserializer;
+        }
+    }
+
+    private final ProcessorTopology topology;
+    private final InternalProcessorContext processorContext;
+    private final Map<TopicPartition, Long> offsets = new HashMap<>();
+    private final Map<String, SourceNodeAndDeserializer> deserializers = new HashMap<>();
+    private final GlobalStateManager stateMgr;
+
+
+    public GlobalStateUpdateTask(final ProcessorTopology topology,
```

Comment:
I disagree. Firstly it isn't mimicing a `ProcessorToplogy` - it is one! We have various classes in place already that do the the work needed to keep the table up-to-date, i.e, `SouceNode` and `KTableSourceProcessor`. They need a `ProcessContext`. If we don't go down this path then we will be duplicating the code in those classes as we need to do the same thing. 
Further `GlobalProcessorContext` is as one-time as `StandbyContextImpl`. I don't see any difference here. I also abstracted out all of the common `ProcessorContext` methods to `AbstractProcessorContext` to avoid having to avoid duplication and implementing of some unnecessary methods.